draft-ietf-core-echo-request-tag.md

```diff
@@ -399,9 +399,9 @@ A single active Echo value with 64 (pseudo-)random bits gives the same theoretic
 
 The security provided by the Echo and Request-Tag options depends on the security protocol used. CoAP and HTTP proxies require (D)TLS to be terminated at the proxies. The proxies are therefore able to manipulate, inject, delete, or reorder options or packets. The security claims in such architectures only hold under the assumption that all intermediaries are fully trusted and have not been compromised.
 
-Servers SHOULD use a monotonic clock to generate timestamps and compute round-trip times. Use of non-monotonic clocks is not secure as the server will accept expired Echo option values if the clock is moved backward. The server will also reject fresh Echo option values if the clock is moved forward. Non-monotonic clocks MAY be used as long as they have deviations that are acceptable given the freshness requirements. If the deviations from a monotonic clock are known, it may be possible to adjust the threshold accordingly.
+Servers SHOULD use a monotonic clock to generate timestamps and compute round-trip times, i.e. a clock source that only increments and never jumps. Use of non-monotonic clocks is not secure as the server will accept expired Echo option values if the clock is moved backward. The server will also reject fresh Echo option values if the clock is moved forward. Non-monotonic clocks MAY be used as long as they have deviations that are acceptable given the freshness requirements. If the deviations from a monotonic clock are known, it may be possible to adjust the threshold accordingly.
```

Comment:
Is "never jumps" a) a necessary property we need, and b) typically a consequence of "monotonic"? (I only know the term's use from maths not clock theory; there "never jumps" would be "continuous" which isn't necessary for "monotonous").

Reply:
Should determine if the clock is continuous or if it ticks. C++ documentation e.g. defines steady_clock as the time between ticks being constant:

Class std::chrono::steady_clock - represents a monotonic clock. The time points of this clock cannot decrease as physical time moves forward and the time between ticks of this clock is constant. This clock is not related to wall clock time (for example, it can be time since last reboot), and is most suitable for measuring intervals.

std::chrono::system_clock - wall clock time from the system-wide realtime clock

Reply:
I remove the definition for the time being. We can put it back in the merged version.